runtime/parachains/src/paras_inherent.rs

```diff
@@ -868,169 +867,243 @@ fn sanitize_backed_candidates<
+			includes_code_upgrade: bool,
 		}
 
-		#[test]
-		fn does_not_truncate_on_exactly_full_block() {
-			new_test_ext(MockGenesisConfig::default()).execute_with(|| {
-				let backed_candidates = vec![BackedCandidate::default()];
-				let max_block_weight =
-					<Test as frame_system::Config>::BlockWeights::get().max_block;
-				// if the consumed resources are precisely equal to the max block weight, we do not truncate.
-				System::set_block_consumed_resources(max_block_weight, 0);
-				assert_eq!(limit_backed_candidates::<Test>(backed_candidates).len(), 1);
-			});
-		}
+		fn make_inherent_data(
+			TestConfig {
+				dispute_statements,
+				dispute_sessions,
+				backed_and_concluding,
+				num_validators_per_core,
+				includes_code_upgrade,
+			}: TestConfig,
+		) -> Bench<Test> {
+			let scenario = BenchBuilder::<Test>::new()
+				.set_max_validators((dispute_sessions.len() as u32) * num_validators_per_core)
+				.set_max_validators_per_core(num_validators_per_core) // 5 validators, 1 validator per core => 5 cores.
+				.set_dispute_statements(dispute_statements)
+				.build(backed_and_concluding, dispute_sessions.as_slice()); // build backed candidates for cores 0 & 1.
 
-		#[test]
-		fn truncates_on_over_full_block() {
-			new_test_ext(MockGenesisConfig::default()).execute_with(|| {
-				let backed_candidates = vec![BackedCandidate::default()];
-				let max_block_weight =
-					<Test as frame_system::Config>::BlockWeights::get().max_block;
-				// if the consumed resources are precisely equal to the max block weight, we do not truncate.
-				System::set_block_consumed_resources(max_block_weight + 1, 0);
-				assert_eq!(limit_backed_candidates::<Test>(backed_candidates).len(), 0);
-			});
+			for c in scenario.data.backed_candidates.iter() {
+				println!("NUMBER OF VALIDITY VOTES {:?}", c.validity_votes.len());
+			}
+			println!("SCENARIO {:?}", scenario.data.disputes.len());
+			for d in scenario.data.disputes.iter() {
+				println!("Amount of statements: {:?}", d.statements.len());
+				println!("SESSION {:?}", d.session);
+			}
+
+			scenario
 		}
 
 		#[test]
-		fn all_backed_candidates_get_truncated() {
+		fn include_backed_candidates() {
 			new_test_ext(MockGenesisConfig::default()).execute_with(|| {
-				let backed_candidates = vec![BackedCandidate::default(); 10];
-				let max_block_weight =
-					<Test as frame_system::Config>::BlockWeights::get().max_block;
-				// if the consumed resources are precisely equal to the max block weight, we do not truncate.
-				System::set_block_consumed_resources(max_block_weight + 1, 0);
-				assert_eq!(limit_backed_candidates::<Test>(backed_candidates).len(), 0);
+				let dispute_statements = BTreeMap::new();
+
+				let mut backed_and_concluding = BTreeMap::new();
+				backed_and_concluding.insert(0, 1);
+				backed_and_concluding.insert(1, 1);
+
+				let scenario = make_inherent_data(
+					TestConfig {
+						dispute_statements,
+						dispute_sessions: vec![0, 0,],
```

Comment:
Why do we want session 0? Isn't the scenario at session 1? Is it just that it does not matter?

Reply:
Yes, sessions are skipped for backed candidates and only applied to disputes. This should be modified to not necessitate the garbage vec entries.